bench/workgen/workgen.cxx

```diff
@@ -2295,13 +2294,49 @@ Workload::run(WT_CONNECTION *conn)
     return (runner.run(conn));
 }
 
+const std::vector<std::string>
+Workload::get_tables()
+{
+    WorkloadRunner runner(this);
+    std::vector<std::string> uris;
+    {
+        const std::lock_guard<std::recursive_mutex> lock(runner._mutex);
+        for (const auto &kv : _context->_internal->_tint) {
+            uris.push_back(kv.first);
+        }
+    }
+    return uris;
+}
+
+void
+Workload::create_table(const std::string &uri)
+{
+    WorkloadRunner runner(this);
+    runner.create_table(uri);
+}
+
 WorkloadRunner::WorkloadRunner(Workload *workload)
     : _workload(workload), _trunners(workload->_threads.size()), _report_out(&std::cout), _start(),
       stopping(false)
 {
     ts_clear(_start);
 }
 
+void
+WorkloadRunner::create_table(const std::string &uri)
+{
+    ContextInternal *icontext = _workload->_context->_internal;
+    {
+        const std::lock_guard<std::recursive_mutex> lock(_mutex);
```

Comment:
The mutex usage is not very clear to me, I am missing something here. We are creating a lock, then passing that as an argument to ```create_all```, but in that function taking another lock on the same mutex?

Are we using the above scheme and hence a recursive mutex because we are incrementing ```_tint_last``` separately to create the table entry, and ideally, we would do them together?